### PR TITLE
games/melonds: Update to 0.9.5

### PR DESCRIPTION
**Pull request by nadiaholmquist**
This updates the melonDS emulator to 0.9.5, which is the latest stable version.

This required backporting a patch which fixes compilation with newer versions of clang, as well as backporting another patch that I just made upstream that fixes a strange issue with the Qt 5 flavor crashing immediately on launch on some configurations, notably FreeBSD and apparently also some Linux systems.

This is my first contribution to ports so I hope I've done everything right, and if there's something upstream could do to make it easier to package for BSD, please do let me know :)

Edit: force pushed an amended commit because I realized I forgot to update the dependencies.

### CODE REVIEW

**Comment by lwhsu** [comment]
cc maintainer @HenryHu

File: games/melonds/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PORTNAME=	melonds
-PORTVERSION=	0.9.4
-PORTREVISION=	1
+PORTVERSION=	0.9.5
+PORTREVISION=	0
```
Comment (lwhsu) [inline games/melonds/Makefile:3]: No need to reset to 0, just remove this line.

**Comment by lwhsu** [comment]
After testing this in the poudriere, it still needs:
  `USES= kde:5`
and
  `USE_KDE= ecm:build`

**Comment by nadiaholmquist** [comment]
That should be those two changes taken care of. However I just remembered that 0.9.5, because of its custom OpenGL workarounds, might need to depend on X11 and Wayland libraries directly.

I'll have to try and set up `poudriere` to make sure I've got this all right.

**Comment by diizzyy** [comment]
Hi,

Please submit this to https://bugs.freebsd.org since we cannot land this as we don't officially support this repo as a way of submitting patches to the ports tree.

Best regards,
Daniel

**Comment by lwhsu** [comment]
merged as 3e10ec461ae5e04573c6180230022e1ccb809f15

**Pull request closed without merging**